lib/format-message.js

```diff
@@ -61,6 +62,11 @@ formatMessage.setup = function setup (opt) {
   if (opt.translate) currentTranslate = opt.translate
   if ('missingReplacement' in opt) missingReplacement = opt.missingReplacement
   if (opt.missingTranslation) missingTranslation = opt.missingTranslation
+  if (opt.customFormats) {
+    if (opt.customFormats.number) number = assign({}, number, opt.customFormats.number)
+    if (opt.customFormats.date) date = assign({}, date, opt.customFormats.date)
+    if (opt.customFormats.time) time = assign({}, time, opt.customFormats.time)
+  }
```

Comment:
Why don't we call the option `formats`, since it will also overwrite built in formats?